ansible/library/gpg_key.py

```
@@ -0,0 +1,116 @@
+                                stderr=subprocess.PIPE,
+                                shell=True)
+        stdout, stderr = proc.communicate()
+        stdout, stderr = stdout.decode(), stderr.decode()
+        retcode = proc.poll()
+        if retcode != 0:
+            output = 'cmd: "%s", code: "%s" stdout: "%s", stderr: "%s"' % (cmd, retcode, stdout, stderr)
+            raise Exception(output)
+        return stdout.rstrip()
+
+    def download_file(self):
+        response = requests.get(self.url, stream=True)
```

Ruff (0.16.3)

[error] Probable use of `requests` call without timeout

(S113)